src/plugins/data_view_field_editor/public/types.ts

```diff
@@ -47,17 +58,49 @@ export interface StartPlugins {
 
 export type InternalFieldType = 'concrete' | 'runtime';
 
+/**
+ * The data model for the field editor
+ * @public
+ */
 export interface Field {
+  /**
```

Comment:
Might make sense to define this in terms of FieldSpec.

Reply:
++

this is a little tricky as the 2 interfaces don't align perfectly. I put together some changes, and testing them in a separate PR in case it gets messy: https://github.com/elastic/kibana/pull/136863/commits/96ebeda0e240eb53d3656ff27bfb584a5b284b12

Reply:
resolved in 9680a199908